### PR TITLE
[new DL] Always use custom property for topbar heights

### LINKED ISSUE #3155: [new DL] Dimmed overlay offset is incorrect when typing in the topbar's search field
In the new DL the topbar height has changed (from 56px to 68px IIRC). That is reflected in the height of the topbar regularly, but when you start typing the in the topbar's search box the greyed out overlay does not start from the new bottom of the topbar, instead it cuts into the topbar, making its content look not vertically centered.

Another giveaway - compare where the grey starts, vs the scrollbar in the left navigation panel - I would expect those to be flush, like when using the original DL.

Screen Shot 2020-08-05 at 6 23 39 PM

It looks like Polaris-TopBar-SearchDismissOverlay's top offset / height should be based upon the new topbar height values, but they are not: https://github.com/Shopify/polaris-react/blob/2015bba82e4d3a9c33c2cfe070347c167ad3499d/src/components/TopBar/components/SearchDismissOverlay/SearchDismissOverlay.scss#L7-L14

Investigating and confirming [other mentions of `top-bar-height`](https://github.com/Shopify/polaris-react/search?q=top-bar-height&unscoped_q=top-bar-height) are adjusting for the new system would also be worthwhile.

### Expected behaviour

The lightened region should match the height of the new topbar, not be smaller than it.

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

Fixes #3155

### WHAT is this pull request doing?

Two commits:

First one fixes a bug:
- Fixes SearchDismissOverlay to use new topbar height token `--p-top-bar-height` so that the grey-ed out overlay is now aligned with the new taller topbar.

Second one does some tidying up:

- Removes usages of the `new-top-bar-height()` sass function and replaces them with references to `--p-top-bar-height`.
- Removes the `new-top-bar-height` sass function - technically a breaking change to the public API but nobody should ever have been using this and we seem to be treating new DL stuff as unstable APIs anyway.

### How to 🎩

Here is a gargantuan Playground that is a mashup of a Frame and Filters example.

To check regressions in the old DL:
- Type into the topbar search box and note that grey overlay aligns with the bottom of the topbar
- Click "More filters" and note that the panel that appears has a heading whose height matches the topbar

To check new functionality - opt into the new DL in storybook then:
- Type into the topbar search box and note that grey overlay aligns with the bottom of the topbar (previously this was short)
- Click "More filters" and note that the panel that appears has a heading whose height matches the topbar (this was previously correct but I've refactored to take advantage of the custom property)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useRef, useState, useCallback} from 'react';
import {
  ArrowLeftMinor,
  HomeMajorMonotone,
  OrdersMajorTwotone,
  ConversationMinor,
} from '@shopify/polaris-icons';

import {
  Page,
  Frame,
  Navigation,
  TopBar,
  Toast,
  ContextualSaveBar,
  Card,
  ActionList,
  Layout,
  FormLayout,
  TextField,
  Modal,
  SkeletonPage,
  TextContainer,
  SkeletonBodyText,
  SkeletonDisplayText,
  Loading,
  ResourceList,
  Filters,
  ChoiceList,
  TextStyle,
  RangeSlider,
  Avatar,
} from '../src';

export function Playground() {
  return <FrameExample />;
}

function FrameExample() {
  const defaultState = useRef({
    emailFieldValue: 'dharma@jadedpixel.com',
    nameFieldValue: 'Jaded Pixel',
  });
  const skipToContentRef = useRef(null);

  const [toastActive, setToastActive] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [isDirty, setIsDirty] = useState(false);
  const [searchActive, setSearchActive] = useState(false);
  const [searchValue, setSearchValue] = useState('');
  const [userMenuActive, setUserMenuActive] = useState(false);
  const [mobileNavigationActive, setMobileNavigationActive] = useState(false);
  const [modalActive, setModalActive] = useState(false);
  const [nameFieldValue, setNameFieldValue] = useState(
    defaultState.current.nameFieldValue,
  );
  const [emailFieldValue, setEmailFieldValue] = useState(
    defaultState.current.emailFieldValue,
  );
  const [storeName, setStoreName] = useState(
    defaultState.current.nameFieldValue,
  );
  const [supportSubject, setSupportSubject] = useState('');
  const [supportMessage, setSupportMessage] = useState('');

  const handleSubjectChange = useCallback(
    (value) => setSupportSubject(value),
    [],
  );
  const handleMessageChange = useCallback(
    (value) => setSupportMessage(value),
    [],
  );
  const handleDiscard = useCallback(() => {
    setEmailFieldValue(defaultState.current.emailFieldValue);
    setNameFieldValue(defaultState.current.nameFieldValue);
    setIsDirty(false);
  }, []);
  const handleSave = useCallback(() => {
    defaultState.current.nameFieldValue = nameFieldValue;
    defaultState.current.emailFieldValue = emailFieldValue;

    setIsDirty(false);
    setToastActive(true);
    setStoreName(defaultState.current.nameFieldValue);
  }, [emailFieldValue, nameFieldValue]);
  const handleNameFieldChange = useCallback((value) => {
    setNameFieldValue(value);
    value && setIsDirty(true);
  }, []);
  const handleEmailFieldChange = useCallback((value) => {
    setEmailFieldValue(value);
    value && setIsDirty(true);
  }, []);
  const handleSearchResultsDismiss = useCallback(() => {
    setSearchActive(false);
    setSearchValue('');
  }, []);
  const handleSearchFieldChange = useCallback((value) => {
    setSearchValue(value);
    setSearchActive(value.length > 0);
  }, []);
  const toggleToastActive = useCallback(
    () => setToastActive((toastActive) => !toastActive),
    [],
  );
  const toggleUserMenuActive = useCallback(
    () => setUserMenuActive((userMenuActive) => !userMenuActive),
    [],
  );
  const toggleMobileNavigationActive = useCallback(
    () =>
      setMobileNavigationActive(
        (mobileNavigationActive) => !mobileNavigationActive,
      ),
    [],
  );
  const toggleIsLoading = useCallback(
    () => setIsLoading((isLoading) => !isLoading),
    [],
  );
  const toggleModalActive = useCallback(
    () => setModalActive((modalActive) => !modalActive),
    [],
  );

  const toastMarkup = toastActive ? (
    <Toast onDismiss={toggleToastActive} content="Changes saved" />
  ) : null;

  const userMenuActions = [
    {
      items: [{content: 'Community forums'}],
    },
  ];

  const contextualSaveBarMarkup = isDirty ? (
    <ContextualSaveBar
      message="Unsaved changes"
      saveAction={{
        onAction: handleSave,
      }}
      discardAction={{
        onAction: handleDiscard,
      }}
    />
  ) : null;

  const userMenuMarkup = (
    <TopBar.UserMenu
      actions={userMenuActions}
      name="Dharma"
      detail={storeName}
      initials="D"
      open={userMenuActive}
      onToggle={toggleUserMenuActive}
    />
  );

  const searchResultsMarkup = (
    <Card>
      <ActionList
        items={[
          {content: 'Shopify help center'},
          {content: 'Community forums'},
        ]}
      />
    </Card>
  );

  const searchFieldMarkup = (
    <TopBar.SearchField
      onChange={handleSearchFieldChange}
      value={searchValue}
      placeholder="Search"
    />
  );

  const topBarMarkup = (
    <TopBar
      showNavigationToggle
      userMenu={userMenuMarkup}
      searchResultsVisible={searchActive}
      searchResultsOverlayVisible
      searchField={searchFieldMarkup}
      searchResults={searchResultsMarkup}
      onSearchResultsDismiss={handleSearchResultsDismiss}
      onNavigationToggle={toggleMobileNavigationActive}
    />
  );

  const navigationMarkup = (
    <Navigation location="/">
      <Navigation.Section
        items={[
          {
            label: 'Back to Shopify',
            icon: ArrowLeftMinor,
          },
        ]}
      />
      <Navigation.Section
        separator
        title="Jaded Pixel App"
        items={[
          {
            label: 'Dashboard',
            icon: HomeMajorMonotone,
            onClick: toggleIsLoading,
          },
          {
            label: 'Jaded Pixel Orders',
            icon: OrdersMajorTwotone,
            onClick: toggleIsLoading,
          },
        ]}
        action={{
          icon: ConversationMinor,
          accessibilityLabel: 'Contact support',
          onClick: toggleModalActive,
        }}
      />
    </Navigation>
  );

  const loadingMarkup = isLoading ? <Loading /> : null;

  const skipToContentTarget = (
    <a id="SkipToContentTarget" ref={skipToContentRef} tabIndex={-1} />
  );

  const actualPageMarkup = (
    <Page title="Account">
      <Layout>
        {skipToContentTarget}
        <Layout.AnnotatedSection
          title="Account details"
          description="Jaded Pixel will use this as your account information."
        >
          <ResourceListFiltersExample />
          <Card sectioned>
            <FormLayout>
              <TextField
                label="Full name"
                value={nameFieldValue}
                onChange={handleNameFieldChange}
              />
              <TextField
                type="email"
                label="Email"
                value={emailFieldValue}
                onChange={handleEmailFieldChange}
              />
            </FormLayout>
          </Card>
        </Layout.AnnotatedSection>
      </Layout>
    </Page>
  );

  const loadingPageMarkup = (
    <SkeletonPage>
      <Layout>
        <Layout.Section>
          <Card sectioned>
            <TextContainer>
              <SkeletonDisplayText size="small" />
              <SkeletonBodyText lines={9} />
            </TextContainer>
          </Card>
        </Layout.Section>
      </Layout>
    </SkeletonPage>
  );

  const pageMarkup = isLoading ? loadingPageMarkup : actualPageMarkup;

  const modalMarkup = (
    <Modal
      open={modalActive}
      onClose={toggleModalActive}
      title="Contact support"
      primaryAction={{
        content: 'Send',
        onAction: toggleModalActive,
      }}
    >
      <Modal.Section>
        <FormLayout>
          <TextField
            label="Subject"
            value={supportSubject}
            onChange={handleSubjectChange}
          />
          <TextField
            label="Message"
            value={supportMessage}
            onChange={handleMessageChange}
            multiline
          />
        </FormLayout>
      </Modal.Section>
    </Modal>
  );

  return (
    <div style={{height: '500px'}}>
      <Frame
        topBar={topBarMarkup}
        navigation={navigationMarkup}
        showMobileNavigation={mobileNavigationActive}
        onNavigationDismiss={toggleMobileNavigationActive}
        skipToContentTarget={skipToContentRef.current}
      >
        {contextualSaveBarMarkup}
        {loadingMarkup}
        {pageMarkup}
        {toastMarkup}
        {modalMarkup}
      </Frame>
    </div>
  );
}

function ResourceListFiltersExample() {
  const [accountStatus, setAccountStatus] = useState(null);
  const [moneySpent, setMoneySpent] = useState(null);
  const [taggedWith, setTaggedWith] = useState(null);
  const [queryValue, setQueryValue] = useState(null);

  const handleAccountStatusChange = useCallback(
    (value) => setAccountStatus(value),
    [],
  );
  const handleMoneySpentChange = useCallback(
    (value) => setMoneySpent(value),
    [],
  );
  const handleTaggedWithChange = useCallback(
    (value) => setTaggedWith(value),
    [],
  );
  const handleFiltersQueryChange = useCallback(
    (value) => setQueryValue(value),
    [],
  );
  const handleAccountStatusRemove = useCallback(
    () => setAccountStatus(null),
    [],
  );
  const handleMoneySpentRemove = useCallback(() => setMoneySpent(null), []);
  const handleTaggedWithRemove = useCallback(() => setTaggedWith(null), []);
  const handleQueryValueRemove = useCallback(() => setQueryValue(null), []);
  const handleFiltersClearAll = useCallback(() => {
    handleAccountStatusRemove();
    handleMoneySpentRemove();
    handleTaggedWithRemove();
    handleQueryValueRemove();
  }, [
    handleAccountStatusRemove,
    handleMoneySpentRemove,
    handleQueryValueRemove,
    handleTaggedWithRemove,
  ]);

  const filters = [
    {
      key: 'accountStatus',
      label: 'Account status',
      filter: (
        <ChoiceList
          title="Account status"
          titleHidden
          choices={[
            {label: 'Enabled', value: 'enabled'},
            {label: 'Not invited', value: 'not invited'},
            {label: 'Invited', value: 'invited'},
            {label: 'Declined', value: 'declined'},
          ]}
          selected={accountStatus || []}
          onChange={handleAccountStatusChange}
          allowMultiple
        />
      ),
      shortcut: true,
    },
    {
      key: 'taggedWith',
      label: 'Tagged with',
      filter: (
        <TextField
          label="Tagged with"
          value={taggedWith}
          onChange={handleTaggedWithChange}
          labelHidden
        />
      ),
      shortcut: true,
    },
    {
      key: 'moneySpent',
      label: 'Money spent',
      filter: (
        <RangeSlider
          label="Money spent is between"
          labelHidden
          value={moneySpent || [0, 500]}
          prefix="$"
          output
          min={0}
          max={2000}
          step={1}
          onChange={handleMoneySpentChange}
        />
      ),
    },
  ];

  const appliedFilters = [];
  if (!isEmpty(accountStatus)) {
    const key = 'accountStatus';
    appliedFilters.push({
      key,
      label: disambiguateLabel(key, accountStatus),
      onRemove: handleAccountStatusRemove,
    });
  }
  if (!isEmpty(moneySpent)) {
    const key = 'moneySpent';
    appliedFilters.push({
      key,
      label: disambiguateLabel(key, moneySpent),
      onRemove: handleMoneySpentRemove,
    });
  }
  if (!isEmpty(taggedWith)) {
    const key = 'taggedWith';
    appliedFilters.push({
      key,
      label: disambiguateLabel(key, taggedWith),
      onRemove: handleTaggedWithRemove,
    });
  }

  return (
    <div style={{height: '568px'}}>
      <Card>
        <ResourceList
          resourceName={{singular: 'customer', plural: 'customers'}}
          filterControl={
            <Filters
              queryValue={queryValue}
              filters={filters}
              appliedFilters={appliedFilters}
              onQueryChange={handleFiltersQueryChange}
              onQueryClear={handleQueryValueRemove}
              onClearAll={handleFiltersClearAll}
            />
          }
          items={[
            {
              id: 341,
              url: 'customers/341',
              name: 'Mae Jemison',
              location: 'Decatur, USA',
            },
            {
              id: 256,
              url: 'customers/256',
              name: 'Ellen Ochoa',
              location: 'Los Angeles, USA',
            },
          ]}
          renderItem={(item) => {
            const {id, url, name, location} = item;
            const media = <Avatar customer size="medium" name={name} />;

            return (
              <ResourceList.Item
                id={id}
                url={url}
                media={media}
                accessibilityLabel={`View details for ${name}`}
              >
                <h3>
                  <TextStyle variation="strong">{name}</TextStyle>
                </h3>
                <div>{location}</div>
              </ResourceList.Item>
            );
          }}
        />
      </Card>
    </div>
  );

  function disambiguateLabel(key, value) {
    switch (key) {
      case 'moneySpent':
        return `Money spent is between $${value[0]} and $${value[1]}`;
      case 'taggedWith':
        return `Tagged with ${value}`;
      case 'accountStatus':
        return value.map((val) => `Customer ${val}`).join(', ');
      default:
        return value;
    }
  }

  function isEmpty(value) {
    if (Array.isArray(value)) {
      return value.length === 0;
    } else {
      return value === '' || value == null;
    }
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Filters/Filters.scss [hunk 1]
```diff
@@ -1,6 +1,6 @@
 @import '../../styles/common';
 
-$list-filters-header-height: rem(56px);
+$list-filters-header-height: top-bar-height();
```
Comment (BPScott) [inline src/components/Filters/Filters.scss:3]: This is the same value, but this makes it clearer that they're related, especially given the mentions of `--p-top-bar-height` below

File: src/components/Frame/Frame.scss [hunk 2]
```diff
@@ -203,7 +203,7 @@ $skip-vertical-offset: rem(10px);
 
 .Main-newDesignLanguage {
   .hasTopBar & {
-    padding-top: new-top-bar-height();
+    padding-top: var(--p-top-bar-height);
```
Comment (BPScott) [inline src/components/Frame/Frame.scss:206]: This class is only ever applied in the new DL so no need for a fallback

File: src/components/Navigation/Navigation.scss [hunk 3]
```diff
@@ -49,7 +49,7 @@ $new-nav-max-width: rem(258px);
 
     .Navigation-newDesignLanguage & {
       display: block;
-      min-height: new-top-bar-height();
+      min-height: var(--p-top-bar-height);
```
Comment (BPScott) [inline src/components/Navigation/Navigation.scss:52]: These classes are only ever applied in the new DL so no need for a fallback

**Comment by BPScott** [comment]
Do bugfixes for new DL stuff require a changelog line?

File: src/components/Filters/Filters.scss [hunk 4]
```diff
@@ -21,39 +21,33 @@ $seperator-color: var(--p-border-subdued, color('sky'));
   width: 100%;
   padding: spacing(base) spacing(loose);
   border-bottom: rem(1px) solid $seperator-color;
-  height: $list-filters-header-height;
+  height: var(--p-top-bar-height, #{$list-filters-header-height});
```
[the remark this answers is not in the document]
Comment (BPScott) [reply src/components/Filters/Filters.scss]: It does not, my paranoia got the best of me! Turns out it's only within `calc` (and [some other functions that are mostly not relevant](https://sass-lang.com/documentation/syntax/special-functions#calc-element-progid-and-expression) that you need to go heavy on the interpolation.

Have updated this and one other place where interpolation was not needed

[the remark this answers is not in the document]
Comment (kyledurand) [reply src/components/Filters/Filters.scss]: Yep `calc()` is weird like that 👍

**Comment by tleunen** [comment]
Out of curiosity, I can see we still have the `top-bar-height()` function and only `new-top-bar-height()` got removed? Any reason to that? Should `top-bar-height()` also be removed or at least deprecated and return `var(--p-top-bar-height)` instead?

**Comment by kyledurand** [comment]
`top-bar-height()` returns a px value which can be used in calculations. Ben pointed out to me when I suggested we return `var(--p-top-bar-height)` from the function that things like [this](https://github.com/Shopify/web/blob/master/app/styles/interop-fixes.scss#L3) would break if we returned a css var instead, so we have to keep it for now

**Comment by tleunen** [comment]
Yeah, that's right, I remember him saying that haha. I think it would help to really mark them as deprecated and removed in next major versions and replace all usage inside Polaris with the variable instead (with fallback on the function)?

And yeah, I was a bit surprised with the breaking change caused by removing the function, but Ben explains the reasons behind it in its initial post, so all good!

**Pull request merged**